platforms/documentation/docs/src/snippets/bestPractices/favorCompositeBuilds-avoid/groovy/buildSrc/src/main/java/org/example/MyPlugin.java

```diff
@@ -0,0 +1,10 @@
+package org.example;
+
+import org.gradle.api.Project;
+import org.gradle.api.Plugin;
+
+public abstract class MyPlugin implements Plugin<Project> {
```

Comment:
🤔 Is there a reason we are not using precompiled script plugins in the example(s)?

Reply:
Is there a reason to do so?  This seems more inline with our best practices, as these plugins and tasks can be better tested (which I'm not doing here for brevity).

Reply:
What do you mean by "can be better tested"?

Reply:
We can write unit test for plugins and tasks written as classes in a full binary plugin vs. a precompiled script plugin.

Reply:
@liutikas - Did you feel strongly about precompiled script plugins?

Reply:
I personally prefer real code instead of precompiled scripts. Java or Kotlin code is familiar to most build engineers, where as precompiled scripts has a varied of Gradle "magic" bits applied that subtly change the behavior of methods like `tasks.register<Foo>("name") { }` where scope changes from `it` to `this`.

In terms of testability, I think writing proper Java/Kotlin does make it easier to write unit tests, as you can pull out separate pieces of build logic and write simple junit tests for it, e.g. to test that a function that your tasks calls works correctly without having to load up all the Gradle machinery.